### PR TITLE
fix huber loss function for when delta = inf

### PR DESCRIPTION
**Pull request by cfifty**
Convert huber loss = l2 loss when delta = inf.

Huber loss should be defined as l2 loss when delta = inf. This clears up poorly defined behavior.

### CODE REVIEW

**Comment by cfifty** [comment]
I've signed it!

File: tensorflow/python/ops/losses/losses_impl.py [hunk 1]
```diff
@@ -419,6 +419,9 @@ def huber_loss(labels, predictions, weights=1.0, delta=1.0, scope=None,
     raise ValueError("predictions must not be None.")
   with ops.name_scope(scope, "huber_loss",
                       (predictions, labels, weights)) as scope:
+    if delta == float('inf'):
```
Comment (alextp) [inline tensorflow/python/ops/losses/losses_impl.py]: This check only works if the delta is a python scalar infinity, not if it's a TF inf float.

Comment (cfifty) [reply tensorflow/python/ops/losses/losses_impl.py]: updated per your comment.

File: tensorflow/python/ops/losses/losses_impl.py [hunk 2]
```diff
@@ -419,6 +420,9 @@ def huber_loss(labels, predictions, weights=1.0, delta=1.0, scope=None,
     raise ValueError("predictions must not be None.")
   with ops.name_scope(scope, "huber_loss",
                       (predictions, labels, weights)) as scope:
+    if not gen_math_ops.is_finite(delta):
```
Comment (alextp) [inline tensorflow/python/ops/losses/losses_impl.py]: This uses the result of a tensor operation as a python boolean, which is only supported inside tf.function / autograph.

As it is right now I believe this code always raises an exception, so please add unit tests to convince yourself the implementation is correct.

Also, why not just modify the huber loss kernel to do the right thing in the presence of infinity there?

**Comment by gbaned** [comment]
@cfifty  Could you please check reviewer comments and keep us posted. Thanks!

**Comment by cfifty** [comment]
> @cfifty Could you please check reviewer comments and keep us posted. Thanks!

Apologies for the delay, ICML's been keeping me busy. I'll update the PR pursuant to Alex's comment this weekend.

File: tensorflow/python/ops/losses/losses_impl.py (hunk 2 above)
Comment (cfifty) [reply tensorflow/python/ops/losses/losses_impl.py]: Great point! Acknowledged.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by cfifty** [comment]
Apologies, it's taking me a while to get bazel setup on my local for testing.

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by gbaned** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!

**Comment by aiswaryasankar** [comment]
**🔒 Entelligence AI Vulnerability Scanner**

✅ **No security vulnerabilities found!**

Your code passed our comprehensive security analysis.

📊 **Files Analyzed**: 1 files

---

**Pull request closed without merging**